### PR TITLE
DBZ-730 add MySqlDecimalIT integration test for all decimal.handling.mode/s

### PR DESCRIPTION
**Pull request by rk3rn3r**
From the time of DBZ-730 we already have many places where we test decimal.handling.mode. But as required I created a separate integration test to cover it.
During development I realized some other things that I need to discuss...

### CODE REVIEW

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    private void assertBigDecimalChangeRecord(SourceRecord record) {
+        Assertions.assertThat(record).isNotNull();
+        final Struct change = ((Struct) record.value()).getStruct("after");
+
+        Assertions.assertThat(change.get("A")).isEqualTo(new BigDecimal("1.33"));
+        Assertions.assertThat(change.get("B")).isEqualTo(new BigDecimal("-2.111"));
+        Assertions.assertThat(change.get("C")).isEqualTo(new BigDecimal("3.44400"));
+        Assertions.assertThat(change.get("D")).isEqualTo(new BigDecimal("15.28000")); // returns the default value
+        Assertions.assertThat(change.getBytes("D")).isNull(); // returns the real value, which is null
```
Comment (rk3rn3r) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java:133]: these 2 lines can be confusing.  especially when getDouble and getString behave differently

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    private void assertStringChangeRecord(SourceRecord record) {
+        Assertions.assertThat(record).isNotNull();
+        final Struct change = ((Struct) record.value()).getStruct("after");
+
+        Assertions.assertThat(change.getString("A").trim()).isEqualTo("1.33");
+        Assertions.assertThat(change.getString("B").trim()).isEqualTo("-2.111");
+        Assertions.assertThat(change.getString("C").trim()).isEqualTo("3.44400");
+        Assertions.assertThat(change.getString("D")).isNull(); // returns the real value, not field's default value
+        // Assertions.assertThat(change.getBytes("D")).isNull(); // seems it's not possible to call getBytes on a String ?
```
Comment (rk3rn3r) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java:161]: getString returns the "real" value, which is null. accessing the field with getBytes() isn't working here.

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+    private void assertDoubleChangeRecord(SourceRecord record) {
+        Assertions.assertThat(record).isNotNull();
+        final Struct change = ((Struct) record.value()).getStruct("after");
+
+        Assertions.assertThat(change.getFloat64("A")).isEqualTo(1.33);
+        Assertions.assertThat(change.getFloat64("B")).isEqualTo(-2.111);
+        Assertions.assertThat(change.getFloat64("C")).isEqualTo(3.44400);
+        Assertions.assertThat(change.getFloat64("D")).isNull(); // returns the real value, not field's default value
+        // Assertions.assertThat(change.getBytes("D")).isNull(); // seems it's not possible to call getBytes on a Double ?
```
Comment (rk3rn3r) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java:147]: getFloat64 returns the "real" value, which is null. accessing the field with getBytes() isn't working here.

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MysqlDefaultValueIT.java [hunk 4]
```diff
@@ -525,7 +526,7 @@ public void realTest() throws InterruptedException {
     public void numericAndDecimalToDoubleTest() throws InterruptedException {
         config = DATABASE.defaultConfig()
                 .with(MySqlConnectorConfig.SNAPSHOT_MODE, MySqlConnectorConfig.SnapshotMode.INITIAL)
-                .with(MySqlConnectorConfig.DECIMAL_HANDLING_MODE, JdbcValueConverters.DecimalMode.DOUBLE)
+                .with(MySqlConnectorConfig.DECIMAL_HANDLING_MODE, RelationalDatabaseConnectorConfig.DecimalHandlingMode.DOUBLE)
```
Comment (rk3rn3r) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MysqlDefaultValueIT.java:529]: isn't this the correct config value?
is it okay to change the parameter with that PR?

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java (hunk 1 above)
Comment (rk3rn3r) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java:133]: Maybe we need to fix BigDecimal handling?

Comment (jpechane) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java:133]: You can use `getWithoutDefault` method

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java (hunk 2 above)
Comment (jpechane) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java:161]: `getBytes` would work only if the schema would be `BYTES` - but it is `String`

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MysqlDefaultValueIT.java (hunk 4 above)
Comment (jpechane) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MysqlDefaultValueIT.java:529]: Yes, that's ok

**Comment by jpechane** [comment]
@rk3rn3r The main problem with this PR is that you mix using `get` whith specific `getXXX` methods. You should either use `get` in all cases or `getXXX` in all cases. Also please take into consideration by remark about `getWithoutDefault` method

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java (hunk 1 above)
Comment (rk3rn3r) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java:133]: oh yes! omg. really, I just learned about that different default value behavior of get(). I would expect the opposite behaviour. get() comes without default and there is getWithDefault()... very interesting.

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java (hunk 2 above)
Comment (rk3rn3r) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java:161]: I still want raw bytes! ;)

**Comment by rk3rn3r** [comment]
Thanks @jpechane! I updated the PR.

**Comment by jpechane** [comment]
@rk3rn3r Applied, thanks!

**Pull request merged**